memory/org.eclipse.cdt.debug.ui.memory.search/src/org/eclipse/cdt/debug/ui/memory/search/MemorySearchResultsPage.java

```diff
@@ -173,39 +176,64 @@ public Object[] getElements(Object inputElement) {
-							.getMemoryRenderingContainers();
-					MemoryMatch match = (MemoryMatch) ((StructuredSelection) event.getSelection()).getFirstElement();
-					if (match != null) {
-						for (int i = 0; i < containers.length; i++) {
-							IMemoryRendering rendering = containers[i].getActiveRendering();
-							if (rendering instanceof IRepositionableMemoryRendering) {
-								try {
-									((IRepositionableMemoryRendering) rendering).goToAddress(match.getStartAddress());
-								} catch (DebugException e) {
-									MemorySearchPlugin.logError(
-											Messages.getString("MemorySearchResultsPage.RepositioningMemoryViewFailed"), //$NON-NLS-1$
-											e);
-								}
-							}
-
-							if (rendering != null) {
-								// Temporary, until platform accepts/adds new interface for setting the selection
-								try {
-									Method m = rendering.getClass().getMethod("setSelection", //$NON-NLS-1$
-											new Class[] { BigInteger.class, BigInteger.class });
-									if (m != null)
-										m.invoke(rendering, match.getStartAddress(), match.getEndAddress());
-								} catch (Exception e) {
-									// do nothing
-								}
-							}
+				IMemoryRenderingContainer containers[] = ((IMemorySearchQuery) fQuery).getMemoryView()
+						.getMemoryRenderingContainers();
+				if (containers == null || containers.length == 0)
+					return;
+
+				if (event.getSelection() instanceof StructuredSelection && ((StructuredSelection) event.getSelection())
+						.getFirstElement() instanceof MemoryMatch match) {
+
+					if (match == null)
+						return;
+
+					for (int i = 0; i < containers.length; i++) {
+						IMemoryRendering rendering = containers[i].getActiveRendering();
+						if (!(rendering instanceof IRepositionableMemoryRendering repositionable))
+							continue;
+
+						try {
+							repositionable.goToAddress(match.getStartAddress());
+						} catch (DebugException e) {
+							MemorySearchPlugin.logError(
+									Messages.getString("MemorySearchResultsPage.RepositioningMemoryViewFailed"), //$NON-NLS-1$
+									e);
+						}
+
+						// Temporary, until platform accepts/adds new interface for setting the selection
+						try {
+							Method m = rendering.getClass().getMethod("setSelection", //$NON-NLS-1$
+									new Class[] { BigInteger.class, BigInteger.class });
+							if (m != null)
+								m.invoke(rendering, match.getStartAddress(), match.getEndAddress());
+						} catch (NoSuchMethodException e) {
+							// Not all renderings have a compatible setSelection
+							// ideally this would be an interface we can check instead
+							// of using reflection!
+						} catch (Exception e) {
+							MemorySearchPlugin.logError(
+									"Exception while invoking the setSelection method for the current rendering", //$NON-NLS-1$
+									e);
 						}
 					}
 				}
 			}
 		});
+		fTreeViewer.addDoubleClickListener(new IDoubleClickListener() {
```

Comment:
Using double-click works well and makes this consistent with other uses of the search view. Thanks!